week-07/day-04/WebApplication1/WebApplication1/Models/ConsoleLoggerMiddleware.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebApplication1.Models
+{
+    public class ConsoleLoggerMiddleware : IMiddleware
+    {
+        //private Printer printer;
+        private Printer printer;
+
+        public ConsoleLoggerMiddleware(Printer printer)
+        {
+            this.printer = printer;
+        }
+
+        public Task InvokeAsync(HttpContext context, RequestDelegate next) // ????????????????????????????????????????
```

Comment:
`InvokeAsync` will be called by .net core for every request and it will pass the context object that contains Request among other things. This way Middleware can be used for some flexible logic, like logging all requests or checking all requests - instead of putting some logic into ALL controller methods (all route handlers)